@shared/api/internal.ts

```diff
@@ -467,6 +481,156 @@ export const getAccountIndexerBalances = async (
+};
+
+export const getAccountBalancesStandalone = async ({
+  publicKey,
+  networkDetails,
+  sorobanClientServer,
+  sorobanClientTxBuilder,
+}: {
+  publicKey: string;
+  networkDetails: NetworkDetails;
+  sorobanClientServer: SorobanRpc.Server;
+  sorobanClientTxBuilder: () => Promise<TransactionBuilder>;
+}): Promise<AccountBalancesInterface> => {
+  const { network, networkUrl, networkPassphrase } = networkDetails;
+
+  let balances: any = null;
+  let isFunded = null;
+  let subentryCount = 0;
+
+  try {
+    const dataProvider = new DataProvider({
+      serverUrl: networkUrl,
+      accountOrKey: publicKey,
+      networkPassphrase,
+      metadata: {
+        allowHttp: networkUrl.startsWith("http://"),
+      },
+    });
+
+    const resp = await dataProvider.fetchAccountDetails();
+    balances = resp.balances;
+    subentryCount = resp.subentryCount;
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < Object.keys(resp.balances).length; i++) {
+      const k = Object.keys(resp.balances)[i];
+      const v: any = resp.balances[k];
+      if (v.liquidity_pool_id) {
+        const server = stellarSdkServer(networkUrl);
+        // eslint-disable-next-line no-await-in-loop
+        const lp = await server
+          .liquidityPools()
+          .liquidityPoolId(v.liquidity_pool_id)
+          .call();
+        balances[k] = {
+          ...balances[k],
+          liquidityPoolId: v.liquidity_pool_id,
+          reserves: lp.reserves,
+        };
+        delete balances[k].liquidity_pool_id;
+      }
+    }
+    isFunded = true;
+  } catch (e) {
+    console.error(e);
+    return {
+      balances,
+      isFunded: false,
+      subentryCount,
+      tokensWithNoBalance: [],
+    };
+  }
+
+  // Get token balances to combine with classic balances
+  const tokenIdList = await getTokenIds(network as NETWORKS);
+
+  const tokenBalances = {} as any;
+  const tokensWithNoBalance = [];
+
+  if (tokenIdList.length) {
+    const params = [new Address(publicKey).toScVal()];
+
+    for (let i = 0; i < tokenIdList.length; i += 1) {
+      const tokenId = tokenIdList[i];
+      /*
+        Right now, Soroban transactions only support 1 operation per tx
+        so we need a builder per value from the contract,
+        once/if multi-op transactions are supported this can send
+        1 tx with an operation for each value.
+      */
+      try {
+        /* eslint-disable no-await-in-loop */
+        const { balance, symbol, ...rest } = await getSorobanTokenBalance(
+          sorobanClientServer,
+          tokenId,
+          {
+            balance: await sorobanClientTxBuilder(),
+            name: await sorobanClientTxBuilder(),
+            decimals: await sorobanClientTxBuilder(),
+            symbol: await sorobanClientTxBuilder(),
+          },
+          params,
+        );
+        /* eslint-enable no-await-in-loop */
+
+        const total = new BigNumber(balance);
+
+        tokenBalances[`${symbol}:${tokenId}`] = {
```

Comment:
here I'm conforming the token balance to the shape the indexer provides so we don't run into type issues elsewhere